src/components/TreesMap/DeckGLMap.tsx

```diff
@@ -600,65 +604,54 @@ class DeckGLMap extends React.Component<DeckGLPropType, DeckGLStateType> {
             height='100%'
           >
             {!showControls && (
-              <>
-                <ControlWrapper isNavOpen={isNavOpen}>
-                  <GeolocateControl
-                    positionOptions={{ enableHighAccuracy: true }}
-                    trackUserLocation={isMobile ? true : false}
-                    showUserLocation={true}
-                    onGeolocate={posOptions => {
-                      const {
-                        coords: { latitude, longitude },
-                      } = (posOptions as unknown) as {
-                        coords: {
-                          latitude: number;
-                          longitude: number;
-                        };
+              <ControlWrapper isNavOpen={isNavOpen}>
+                <GeolocateControl
+                  positionOptions={{ enableHighAccuracy: true }}
+                  trackUserLocation={isMobile ? true : false}
+                  showUserLocation={true}
+                  onGeolocate={posOptions => {
+                    const {
+                      coords: { latitude, longitude },
+                    } = (posOptions as unknown) as {
+                      coords: {
+                        latitude: number;
+                        longitude: number;
                       };
-                      this.setViewport({
-                        longitude,
-                        latitude,
-                        zoom: VIEWSTATE_ZOOMEDIN_ZOOM,
-                        transitionDuration: VIEWSTATE_TRANSITION_DURATION,
-                      });
-                    }}
-                  />
-                  <NavigationControl
-                    onViewStateChange={e =>
-                      this.setViewport({
-                        latitude: e.viewState.latitude,
-                        longitude: e.viewState.longitude,
-                        zoom: e.viewState.zoom,
-                        transitionDuration: VIEWSTATE_TRANSITION_DURATION,
-                      })
-                    }
-                  />
-                </ControlWrapper>
-                {!isMobile &&
-                  hoveredPump &&
-                  hoveredPump.latitude &&
-                  hoveredPump.longitude && (
-                    <Popup
```

Comment:
What is the reason for you not being able to reuse `Popup` here. You had to recreate all this logic again, although the problem only seems to be related to the order in which GeoJSON/React layers are rendered on the page. What pushed you to make this decision? Were there specific issues, that forced you to create a new `Tooltip` component?

Reply:
The thing is that DeckGL lays a `canvas` on top of react-map-gl's own canvas. This is necessary since we want our points to be rendered on top of the base map etc. Additionally, as far as I know, we can't change the order since we need to include react-map-gl via `<StaticMap>` [here](https://github.com/technologiestiftung/giessdenkiez-de/blob/a0394aee36e7fd22c4a513b82e0f633ae9db3fd1/src/components/TreesMap/DeckGLMap.tsx#L593) as a child of DeckGL. `<Popup>` belongs to react-map-gl so it is forced to be rendered below.

The fix moves the tooltip outside of the react-map-gl context, that's why it was necessary to create a dedicated tooltip component again.